openapi3/discriminator.go

```diff
@@ -9,9 +9,10 @@ import (
 // See https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#discriminator-object
 type Discriminator struct {
 	Extensions map[string]any `json:"-" yaml:"-"`
+	Origin     *Origin        `json:"origin,omitempty" yaml:"origin,omitempty"`
 
-	PropertyName string            `json:"propertyName" yaml:"propertyName"` // required
-	Mapping      map[string]string `json:"mapping,omitempty" yaml:"mapping,omitempty"`
+	PropertyName string    `json:"propertyName" yaml:"propertyName"` // required
+	Mapping      StringMap `json:"mapping,omitempty" yaml:"mapping,omitempty"`
```

Comment:
Please move the introduction of the `StringMap` type to another PR, with a breaking change note to README if applicable.

Reply:
Done: https://github.com/getkin/kin-openapi/pull/1018
After https://github.com/getkin/kin-openapi/pull/1018 is merged, I will update this one to leverage it.

Reply:
Yes, I was struggling but managed to do it eventually. Thanks.